test/test_service.rb

```diff
@@ -273,4 +273,66 @@ def test_patch_service
       data['metadata']['annotations']['key'] == 'value'
     end
   end
+
+  def test_json_patch_service
+    service = Kubeclient::Resource.new
+    name = 'my_service'
+
+    service.metadata = {}
+    service.metadata.name      = name
+    service.metadata.namespace = 'development'
+
+    stub_core_api_list
+    expected_url = "http://localhost:8080/api/v1/namespaces/development/services/#{name}"
+    stub_request(:patch, expected_url)
+      .to_return(body: open_test_file('service_patch.json'), status: 200)
+
+    patch = {
+      metadata: {
+        annotations: {
+          key: 'value'
+        }
+      }
+    }
```

Comment:
JSON patch takes different format — list of {op, path, value} tuples.  https://tools.ietf.org/html/rfc6902
I think here you want:
```
patch = [
  {op: "replace", path: "/metadata/annotations/key", value: "value"}
]
```